### ISSUE TITLE
Cherrytree not opening encrypted file.

### ISSUE THREAD

**Issue by HenkoBeukes** [giuspen/cherrytree#2108]
Cherrytree version 0.99.48 on Ubuntu 22.04.1 LTS
ppa version and snap version

After update: Cannot open a file (.ctx) with the password. The file is not corrupted as none of the previously verified backups can be opened either. My password is very basic with capitals, lowercase, numbers and $.
I initially had the snap version but removed that version when I encountered the problem and installed the ppa version. Problem persisted with slight difference.

snap version: 
Trying to open a file from the menu "open file" tab in the menu:  the app closes without action. 
Using the "recent documents" tab  in menu: brings up the password window but it just clears the window when clicking on "OK". No action or error message. 

ppa version: "open file " tab opens the "select file" window,  but same as before it does not take the password when entered, just clears the entered password. 

The app seems to work with an unencrypted file.

**Comment by giuspen** [comment]
The snap is known to have had a glitch but the ppa has never had a reported issue on the encrypted files.
I suspect the real issue is the jump from the snap version which was 0.39.4 (showing 0.99.36 for a mistake) to 0.99.48.
I remember the old 0.39.4 had an issue with the $ character.
If you keep failing I'll help you extract your data but first try the following:
if your password is for example one$two try to enter instead in the new 0.99.48:
onetwo
onewo
one
Let me know

**Comment by giuspen** [comment]
If you keep failing, temporarily install the old snap marked 0.99.36 (which is in fact the old python version 0.39.4) which is still available in the channel edge:

giuspen@mint2204:~$ snap info cherrytree
name:      cherrytree
...
channels:
  latest/stable:    0.99.48 2022-08-26 (56) 247MB -
  latest/candidate: ↑                             
  latest/beta:      ↑                             
  latest/edge:      0.99.36 2022-08-26 (50)  71MB -

Then after opening the encrypted document save it as unencrypted, switch back to latest 0.99.48 and finally encrypt it again

**Comment by HenkoBeukes** [comment]
I managed to open my file by omitting all characters from the $ in my password. Then saved it as unencrypted as I can't lose 15Mb of notes. 
Thanks for the help.  Great app!

**Comment by DanteJoseph** [comment]
I recently installed the latest update, 1.1.2 and now can't open my encrypted file :(
I enter the correct password repeatedly and the enter password dialog box just keeps reappearing.
Please help!

**Comment by giuspen** [comment]
Hi, if you didn't disable the backups you will have 3 time based backups in the same folder of the original document. Same name of the original but a tilde at the end (or 2 or 3). Get the one with one tilde, copy in another folder, rename removing the trailing tilde, try to open that.
E.g. myfile.ctx~ -> myfile.ctx
You can also try to open directly with https://www.7-zip.org/ as that is the tool used to encrypt and password protect.
In the next version v1.1.3 I'm implementing auto recovery from the backups in case of corruption

**Comment by DanteJoseph** [comment]
Thank you for your prompt attention!  I tried doing what you recommended and found the backup files.  I went to the most recent backup file, which was last Friday (5/17), and copied the file to a different location.  I renamed it to remove the tilde and changed the file name to today and also removed the dot in the front, but still have the same problem.  
- Old file name: .DanteJ_2024-0214enc.ctx~ 
- New file name: DanteJ_2024-0521.ctx
![Screenshot 2024-05-21 102123](https://github.com/giuspen/cherrytree/assets/4978426/36ca16ee-5517-4924-99c3-650e16ee4315)
-
![Screenshot 2024-05-21 102221](https://github.com/giuspen/cherrytree/assets/4978426/c4e9acc6-36a4-4294-ac24-7ba4724fb75a)

Is it possible I should have left the dot in front of the file name or should I try an older date of backup file?
I do not know what to do regarding trying to open directly with https://www.7-zip.org/ I did download that, but do not know what to do from there.

**Comment by giuspen** [comment]
@DanteJoseph which cherrytree version version were you using before v1.1.2?
Did you ever successfully open with v1.1.2?
In recent versions before rotating the backups it is opening and running a basic parsing in the background so I'm a bit puzzled. If you want you can try to send me privately a document and write me the password I can give it a try too, my email is in help--about dialog

**Comment by DanteJoseph** [comment]
@giuspen Thanks again for your continued support.  I was using version 1.1.1 before the update, and I was able to open the file with the updated version, but after a PC reboot, this issue began.  I wish I could send you the file and password, but it contains quite a bit of personal information.  Any other suggestions?  Is there an updated version in the works and, if so, do you think it would resolve this issue?

**Comment by giuspen** [comment]
@DanteJoseph the first thing to understand here is if the compressed and password protected archive is corrupted or the password is incorrect.
In order to understand that, you should try and open the .ctx file with https://www.7-zip.org/ that will tell you if it is not recognised as a 7zip Archive or will ask you for the password

**Comment by giuspen** [comment]
@DanteJoseph unfortunately yes, the binary file is not recognised as a 7zip archive which means it is corrupted.
I'm puzzled as a backup is not rotated before being extracted and re-parsed, exactly to avoid this situation.
I will spend some time on testing/analysing trying to figure out what could have happened, unfortunately I don't see a way to recover your data.

**Comment by DanteJoseph** [comment]
@giuspen Yeah, I recall my PC was frozen the other day and I had to do a forced shut down, so I am guessing that caused the issue, but I agree it is weird that the backups won't restore previous version saved...
Thank you for all your efforts!
